pkg/sentry/socket/plugin/cgo/socket_unsafe.go

```diff
@@ -0,0 +1,445 @@
+// Copyright 2023 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cgo
+
+/*
+#include <stdint.h>
+#include <sys/epoll.h>
+#include <sys/socket.h>
+
+// epoll
+int plugin_epoll_create(void);
```

Comment:
Where are all of these C functions implemented?

Reply:
These C functions are implemented in TLDK: https://github.com/alipay/tldk/blob/main/lib/libtle_glue/gvisor_glue_sockops.c